### PR TITLE
feat: Add `features_always_increment_minor` flag

### PR DESCRIPTION
**Pull request by scouten-adobe**
When enabled (off by default), it triggers a minor version bump for `feat` commits even when the baseline version is 0.x.

This implements the feature requested in #1656.

### CODE REVIEW

**Comment by scouten-adobe** [comment]
There's a Clippy failure, but that is in code I haven't changed in  this PR. Rust 1.81.0 was released today and this Clippy failure does not appear when I backtrack to 1.80.1, so I'll leave it out of this PR.

File: crates/release_plz/src/config.rs [hunk 1]
```diff
@@ -264,6 +264,10 @@ pub struct PackageConfig {
     /// Whether to create/update changelog or not.
     /// If unspecified, the changelog is updated.
     pub changelog_update: Option<bool>,
+    /// # Features Always Increment Minor Version Flag
```
Comment (MarcoIeni) [inline crates/release_plz/src/config.rs:267]:
```suggestion
    /// # Features Always Increment Minor Version
```
the "flag" suffix is not present anywhere else.

**Comment by MarcoIeni** [comment]
Can you update the docs?

**Comment by scouten-adobe** [comment]
@MarcoIeni happy to do so. Will look at it over the weekend.

**Comment by scouten-adobe** [comment]
@MarcoIeni updated the docs as requested.

I searched all 39 instances of `test]` that were present in `crates/release-plz/**.rs` but did not find an existing example of where a new version was chosen.

Could you point me to a specific existing example of the inspiration you were hoping I would follow? I'm happy to adapt an existing example, but I'm not sure how to start fresh.

**Comment by MarcoIeni** [comment]
I think this is a good starting point:
https://github.com/MarcoIeni/release-plz/blob/f818f0b1262131d33c31ec53a322f044e307ed64/crates/release_plz/tests/all/release_pr.rs#L8

**Comment by MarcoIeni** [comment]
there's a git conflict, can you resolve it?

**Comment by scouten-adobe** [comment]
@MarcoIeni I think I've caught up now

**Comment by MarcoIeni** [comment]
Let me know if you need help with the failing test 👍

**Comment by scouten-adobe** [comment]
@MarcoIeni the failure mode looks identical to the one that I drafted from, so I copied your `#[ignore]` language as well.

I did confirm it working locally on my Mac. I'm not sure where to begin on fixing the message.

**Comment by MarcoIeni** [comment]
oh, I saw that the other test doesn't work on CI either 😅
I'll merge this 👍

**Comment by scouten-adobe** [comment]
@MarcoIeni thank you. Great working with you on this!

**Comment by MarcoIeni** [comment]
same, I hope to see you again in my gh notifications :D

**Review by MarcoIeni — APPROVED** [review]

**Pull request merged**